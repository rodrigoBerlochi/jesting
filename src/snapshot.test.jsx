import React from 'react';
import renderer from 'react-test-renderer';

const Link = (url, text) => { 
    return <a href={url}>text</a>; 
}

describe.only('Snapshot', () => {
    test('Basic snap', () => {
        const tree = renderer.create(
            <Link url="http://google1.com.es" text="hello!" />
        ).toJSON();

        expect(tree).toMatchSnapshot();
        // recall INTERACTIVE update mode in watch
    });

    test('Basic snap 2', () => {

        const lemonTree = renderer.create(
            <p>A yellow submarine!</p>
        ).toJSON();

        expect(lemonTree).toMatchSnapshot();
    });

    test('Asymmetric matchers', () => {
        const v = {
            createdAt: new Date(),
            id: Math.floor(Math.random() * 100),
            name: 'John'
        };

        // createdAt and Id will change on each run, so
        // tell Jest how to test them

        expect(v).toMatchSnapshot({
            createdAt: expect.any(Date),
            id: expect.any(Number)
        });
    });

    /**
     * Good practices
     * - snapshots are code. Commits, review. 
     * - be deterministic, mock what isn't so
     * - names describe expected result
     */
})