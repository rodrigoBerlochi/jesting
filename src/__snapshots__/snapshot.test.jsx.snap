// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Basic snap 1`] = `
<a
  href={
    Object {
      "text": "hello!",
      "url": "http://google1.com.es",
    }
  }
>
  text
</a>
`;

exports[`Snapshot Basic snap 2 1`] = `
<p>
  A yellow submarine!
</p>
`;
